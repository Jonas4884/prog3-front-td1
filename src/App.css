:root {
	--omrs-color-ink-lowest-contrast: rgba(47, 60, 85, 0.18);
	--omrs-color-ink-low-contrast: rgba(60, 60, 67, 0.3);
	--omrs-color-ink-medium-contrast: rgba(19, 19, 21, 0.6);
	--omrs-color-interaction: #1e4bd1;
	--omrs-color-interaction-minus-two: rgba(73, 133, 224, 0.12);
	--omrs-color-danger: #b50706;
	--omrs-color-bg-low-contrast: #eff1f2;
	--omrs-color-ink-high-contrast: #121212;
	--omrs-color-bg-high-contrast: #ffffff;
	
}