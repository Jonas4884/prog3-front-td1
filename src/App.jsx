import { useState } from 'react';
import TodoItemList from './common/TodoList';
import TodoForm from './common/TodoForm';
import { TodoStatus } from './common/utils/status';
import './assets/index1.css';


function App() {
  const [todos, setTodos] = useState([]);

  const addTodo = (newTodo) => {
    setTodos(currentTodos => [...currentTodos, newTodo]);
  };

  const removeTodo = (todoId) => {
    setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
  };

  const doneTodo = (todoId) => {
    const editTodos = (currentTodos) => {
      return currentTodos.map(todo => (todo.id !== todoId ? todo : { ...todo, status: TodoStatus.DONE }));
    };
    setTodos(editTodos);
  };

  return (
    <div className='container__around'>
      <div className='todo__list__content'>
        <TodoItemList actions={<TodoForm add={addTodo} />} todoStatus={TodoStatus.TODO} todos={todos} onDone={doneTodo} />
        <TodoItemList todoStatus={TodoStatus.DONE} todos={todos} onDone={doneTodo} onRemove={removeTodo} />
        <p>{
          
          JSON.stringify(todos)
          }</p>
      </div>
    </div>
  );
}

export default App;
