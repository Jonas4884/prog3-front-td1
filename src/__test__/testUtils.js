export const todo = [{"id":1675425781448,"status":"done","value":"task 1"},{"id":1675425784921,"status":"done","value":"task 2"},{"id":1675425789408,"status":"todo","value":"task 3"},{"id":1675425831658,"status":"todo","value":"task 4"}]
export const done = [{"id":1675422371749,"status":"done","value":"clscnjsbcmks"},{"id":1675422384471,"status":"done","value":"csjcsb"},{"id":1675422586728,"status":"done","value":"cldcd"}]