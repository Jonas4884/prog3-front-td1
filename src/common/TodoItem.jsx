import { TodoStatus } from "./utils/status";


const TodoItem = (props) => {
  const { onDone, todo, onRemove } = props;

  const toDone = () => {
    onDone && onDone(todo.id);
  };

  const remove = () => {
    onRemove && onRemove(todo.id);
  };

  return (
    <div className='todo__item todo__value' role={"the__item"}>
      <input name="todo" type='checkbox' className='pointer' onClick={toDone} checked={todo.status === TodoStatus.DONE} id={todo.id.toString()} />
      <label htmlFor="todo"  className='item__value' role={"todovalue"}>{todo.value}</label>
      {todo.status === TodoStatus.DONE && (
        <p className='pointer'  role={"remove_area"} onClick={remove}>
          x
        </p>
      )}
    </div>
  );
};

export default TodoItem;
