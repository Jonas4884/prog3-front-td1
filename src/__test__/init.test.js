import { render, screen } from "@testing-library/react";
import TodoItem from "../common/TodoItem";
import { done, todo } from "./testUtils";
import renderer from 'react-test-renderer'
import userEvent from "@testing-library/user-event";
import App from "../App";
import { useState } from "react";

describe('should return main interface', () => {
    test('should return title TODO', () => { 
        render(<App/>);
        const titleTorender = screen.getByText("TODO");
    expect(titleTorender).toBeInTheDocument();
     })

    test('should return title DONE', () => { 
        render(<App/>);
        const titleTorender = screen.getByText("DONE");
    expect(titleTorender).toBeInTheDocument();
     })
     test('should match the initiate snapshot', () => { 
        const renderComponent = renderer.create(<App/>).toJSON();
        expect(renderComponent).toMatchSnapshot()
      })

     test("should remove item onClick", () => {
        useState = jest.fn()
        const [todos, setTodos] = useState([]);
        const removeTodo = (todoId) => {
            setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
          };
        render(<TodoItem todo={todos.at(0)} onRemove={removeTodo} />);
        const actualStatus = screen.getByRole("remove_area");
        userEvent.click(actualStatus);
        expect(done.at(0).value).not.toBeInTheDocument();
      });
});