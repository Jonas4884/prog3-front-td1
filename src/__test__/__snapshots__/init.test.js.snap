// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return main interface should match the initiate snapshot 1`] = `
<div
  className="container__around"
>
  <div
    className="todo__list__content"
  >
    <div
      className="todo__array"
      role="todo__array"
    >
      <h1
        title="todo__title"
      >
        TODO
      </h1>
      <div
        className="actions"
        role="application"
      >
        <form
          onSubmit={[Function]}
          test-id="the__item"
        >
          <input
            className="todo__input"
            data-testid="element-to-focus"
            id="name"
            name="name"
            onChange={[Function]}
            placeholder="Add new task"
            type="text"
            value=""
          />
          <label
            className="form__label"
            htmlFor="name"
          >
            Press enter to add Task
          </label>
        </form>
      </div>
      <div>
        <p>
          No task
        </p>
      </div>
    </div>
    <div
      className="todo__array"
      role="todo__array"
    >
      <h1
        title="todo__title"
      >
        DONE
      </h1>
      <div
        className="actions"
        role="application"
      />
      <div>
        <p>
          No task
        </p>
      </div>
    </div>
    <p>
      []
    </p>
  </div>
</div>
`;
